staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -478,7 +487,11 @@ func index(l []interface{}, valToLookUp interface{}, mergeKey string, kind refle
 	}
 
 	for i, v := range l {
-		if getValFn(valToLookUp) == getValFn(v) {
+		left, right := getValFn(valToLookUp), getValFn(v)
```

Comment:
Seems like you could move the initialization of `left` outside of the loop. But the compiler will probably do that automatically.